consensus/src/consensus/consensus_proxy.rs

```diff
@@ -239,33 +254,51 @@ impl<N: Network> ConsensusProxy<N> {
                     continue;
                 }
 
-                if block_number <= &election_head.block_number() {
-                    // First Case: Transactions from finalized epochs
-                    hashes_by_block
-                        .entry(Policy::election_block_after(*block_number))
-                        .or_insert(vec![])
-                        .push(hash.clone());
-                } else if block_number <= &checkpoint_head.block_number() {
-                    // Second Case: Transactions from a finalized batch in the current epoch
-                    hashes_by_block
-                        .entry(checkpoint_head.block_number())
-                        .or_insert(vec![])
-                        .push(hash.clone());
+                // There are essentially two different cases that we need to handle
+                if let Some(block_number) = block_number {
+                    // Case A: We are provided a block number, so we need to determine which is the best proving block
+                    // There are three sub-categories of block numbers:
+                    //  - Finalized epochs: we use the election block number that finalized the respective epoch
+                    //  - Finalized batch in the current epoch: We use the latest checkpoint block number
+                    //  - Current batch: We use the current head to prove those transactions
+                    if block_number <= &election_head.block_number() {
+                        // First Case: Transactions from finalized epochs
+                        hashes_by_block
+                            .entry(Some(Policy::election_block_after(*block_number)))
+                            .or_insert(vec![])
+                            .push(hash.clone());
+                    } else if block_number <= &checkpoint_head.block_number() {
+                        // Second Case: Transactions from a finalized batch in the current epoch
+                        hashes_by_block
+                            .entry(Some(checkpoint_head.block_number()))
+                            .or_insert(vec![])
+                            .push(hash.clone());
+                    } else {
+                        // Third Case: Transanctions from the current batch
+                        hashes_by_block
+                            .entry(Some(current_head.block_number()))
+                            .or_insert(vec![])
+                            .push(hash.clone());
+                    }
                 } else {
-                    // Third Case: Transanctions from the current batch
+                    //Case B: We are not provided a block_number
                     hashes_by_block
-                        .entry(current_head.block_number())
+                        .entry(None)
```

Comment:
Haha, that is possible? An entry for `None`? lol :D

Reply:
Yes :) , they key type is an Option